pkg/proto/datadog/trace/stats.proto

```diff
@@ -49,6 +49,10 @@ message ClientStatsPayload {
 	string git_commit_sha = 13;
 	// The image tag is obtained from a container's set of tags.
 	string image_tag = 14;
+        // The process tags hash is used as a key for agent stats agregation.
+        uint64 process_tags_hash = 15;
+        // The process tags contains a list of tags that are specific to the process.
+        string process_tags = 16;
```

Comment:
why not have a `repeated string` field with split tags, same as container tags? This way you don't have to split the tags when computing the hash.

Reply:
I was also going to suggest using camelCase for consistency but noticed we already have a mix of cases :(

Reply:
since we're propagating this payload to multiple intakes, to keep the same normalisation I preferred to not touch what's provided by the tracing library and normalize in the backend
Tracer library will have it's own normalisation at the source

Reply:
> This way you don't have to split the tags when computing the hash.

I think this can actually be a pretty big deal when it comes to performance, so we should really think twice about not doing this and be very intentional about it. It's probably fine, but it would be nice to understand what perf penalty we might be eating here with this decision.

Reply:
What performance cost do you expect? This should be actually the most efficient
1. tracers normalise once their process tags in their lifetime
2. trace-agent is a passthrough on them
3. backends have a second normalization to cover further renormalization

Reply:
So, maybe I'm missing something but from this comment:

>This way you don't have to split the tags when computing the hash.

The associated cost I'm alluding to is related to that precisely, the need to do string allocations due to the string split, and then calculating the hash on that split.

Reply:
that split would still be needed if we stored this as an array of string here, the process tags are received through a http header (string to split)